task/buildah-remote/0.1/buildah-remote.yaml

```diff
@@ -198,6 +209,12 @@ spec:
         echo "Prefetched content will be made available"
       fi
 
+      reposdir_path=$(realpath $SOURCE_CODE_DIR)/$REPOSDIR_SRC
```

Comment:
The overall idea is to allow the user to define yum repositories to use during the build process. The user can either provide it together with the source code or to have it generated by another task.

Having the directory inside  the `source` dir makes it easier for the user to provide their own yum repositories and having them maintained together with the code. Moving it out will mean that the user will need to have it stored on another repo, as far as I see it.

Reply:
I think what we can do is to use the same workspace, but not the `source` dir in it.

We'll mount a directory (`repos.d`) on that workspace to `/etc/yum.repos.d` (or a specified target).
This directory will be at the same level as the `source` dir where the content of the git repo is.

The pre-build task we'll push content to that directory, and the buildah task finds content on `source/$REPOSDIR_SRC`, it will copy this under the `repos.d` directory.

This way, the user won't have to move things around, but we also don't risk making changes inside the directory pulled from git.

Reply:
That sounds reasonable :+1:

Reply:
@chmeliik can you have a look to see if the changes address your concerns?

Reply:
They do :+1: 

In fact, this even solves https://github.com/redhat-appstudio/build-definitions/pull/724#discussion_r1445846933, since we'll be mounting a copy of the repo files, we won't be mounting them directly from the source dir

Reply:
Though eventually, we could consider moving the copying of `YUM_REPOS_D_SRC` to the task that handles repos - IMO buildah shouldn't really have to know about both the locations (the one to mount from and the one to copy from)

Reply:
> we could consider moving the copying of YUM_REPOS_D_SRC to the task that handles repos

That task will not be used in case yum repos are only taken from the source code repo itself.
Suppose that we move it outside of the buildah task, we'll need to have another task to prep the mount point in a similar manner to what we're doing here now.

Reply:
@chmeliik do you see this as a blocker for proceeding with the current PR? We plan on doing a demo tomorrow and it will be super helpful if we can get it through.

Reply:
Not a blocker, just something to potentially clean up for the RHTAP (former Dance) version of the buildah task